File: backend/src/main/java/com/apocalypse/thefall/mapper/MapMapper.java
package com.apocalypse.thefall.mapper;

import com.apocalypse.thefall.dto.MapDto;
import com.apocalypse.thefall.entity.Map;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring", uses = TileMapper.class)
public interface MapMapper {
    MapDto toDto(Map map);
}

File: backend/src/main/java/com/apocalypse/thefall/mapper/MapMapper.kt
package com.apocalypse.thefall.mapper

import com.apocalypse.thefall.dto.MapDto
import com.apocalypse.thefall.entity.GameMap

fun GameMap.toDto(): MapDto =
    MapDto(
        id = this.id,
        name = this.name,
        width = this.width,
        height = this.height,
    )